### ISSUE TITLE
Fatal error: Cannot redeclare GeorgRinger\News\Domain\Model\News upon install of any extension

### ISSUE THREAD

**Issue by rafu1987** [georgringer/news#1576]
## Bug Report

**Current Behavior**
When installing (or updating) any extension, I'm getting this error in the backend:
Fatal error: Cannot redeclare GeorgRinger\News\Domain\Model\News

I stuck to this guide:
https://docs.typo3.org/p/georgringer/news/master/en-us/DeveloperManual/ExtendNews/ProxyClassGenerator/Index.html

But my extended Model gets loaded twice, when installing any extension. After clearing the cache, the error is gone and the cache file is correctly generated.

**Expected behavior/output**
No error message.

**Environment**
- TYPO3 version(s): 10.4.19
- news version: 8.6.0
- Is your TYPO3 installation set up with Composer (Composer Mode): no
- OS: OSX 11.5.2

**Comment by DavidBruchmann** [comment]
Looks like you used the original namespace in your own extension.
You have to use your own one and extend the original one:
```
namespace [VENDOR]\[EXTENSION]\Domain\Model;

class News extends \GeorgRinger\News\Domain\Model\News {
    
}
```
Beside that it might save you from some hassle if you keep opening curly brackets in the same line with class and method-declaration like shown above.

**Comment by rafu1987** [comment]
@DavidBruchmann nope - that's my namespace:

`namespace RZ\Razornews\Domain\Model;`

and I am extending like this:

`class News extends \GeorgRinger\News\Domain\Model\News`

like you said.

So overall, I actually did extend news exactly the way it is described in the manual, but I'm still getting the error upon installing any new extension in non-composer mode.

Here's the complete error, which is thrown in backend after installing any extension:

`Fatal error: Cannot redeclare GeorgRinger\News\Domain\Model\News::$foobar in /www/ulivinum/typo3temp/var/cache/code/news/tx_news_domain_model_news.php on line 1759`

After clearing the cache, the news cache file is correctly generated. If I look into the cache file right after the error is thrown, I can see that my extend model is included twice.

**Comment by DavidBruchmann** [comment]
mhm, without seeing more code I can only guess wildly.
Feel free to send me the extension by mail if you never want to publish it, then I'll have a look.

**Comment by rafu1987** [comment]
@DavidBruchmann thanks for the offer, I'll send it to you via email, as it is part of a distribution. I hope it'll work without the distribution, but normally it should. :-)

**Comment by rafu1987** [comment]
I debugged `$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']` in file ClassCacheManager in the reBuild function. This is the result, although I only added the classes once to

`$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/News'][]
`
and once to

`$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/Category'][]`

```
array(2) {
  ["Domain/Model/News"]=>
  array(2) {
    [0]=>
    string(9) "razornews"
    [1]=>
    string(9) "razornews"
  }
  ["Domain/Model/Category"]=>
  array(2) {
    [0]=>
    string(9) "razornews"
    [1]=>
    string(9) "razornews"
  }
}
```

This only happens when installing an extension (almost every time) and the mentioned error is thrown. I checked everywhere, but couldn't find anything:

<img width="238" alt="Bildschirmfoto 2021-09-20 um 17 55 45" src="https://user-images.githubusercontent.com/2694825/134033977-293fa55e-b22e-4608-b482-2b2085a59c26.png">

**Comment by DavidBruchmann** [comment]
I doubt that you need `$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/Category'][]`.
Honestly I don't know, but you should try to deactivate it.

**Comment by DavidBruchmann** [comment]
and even if you need it, the value should be different than for 'Domain/Model/News'

**Comment by DavidBruchmann** [comment]
$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/News'][] = 'razornews';
$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/Category'][] = 'razornewscategory';

**Comment by rafu1987** [comment]
> and even if you need it, the value should be different than for 'Domain/Model/News'

That is not correct as the value is always the extension key - this is also mentioned in the manual!

**Comment by DavidBruchmann** [comment]
is it possible that've the notation in ext_tables.php as well as in Configutration/TCA/Overrides/tx_news_domain_model_news.php ?

**Comment by DavidBruchmann** [comment]
Actually the initial posted error-message is not about a double declared class, but about a twice declared variable.
So thanks about the hint to read more thoroughly 😆

**Comment by rafu1987** [comment]
I kind of solved it, by adding these lines:

```
$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/News']['razornews'] = 'razornews';
$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/Category']['razornews'] = 'razornews';
```

Before I had set (mentioned in the documentation):

```
$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/News'][] = 'razornews';
$GLOBALS['TYPO3_CONF_VARS']['EXT']['news']['classes']['Domain/Model/Category'][] = 'razornews';
```

Now, my model is not added twice to the news cache file.
